Core/clim-core/table-formatting.lisp

```diff
@@ -294,31 +332,55 @@ skips intervening non-table output record structures."))
         (if multiple-columns-x-spacing
             (parse-space stream multiple-columns-x-spacing :horizontal)
             x-spacing))
-  (with-new-output-record
-      (stream record-type table
-              :x-spacing x-spacing
-              :y-spacing y-spacing
-              :multiple-columns multiple-columns
-              :multiple-columns-x-spacing multiple-columns-x-spacing
-              :equalize-column-widths equalize-column-widths)
-    (multiple-value-bind (cursor-old-x cursor-old-y)
-        (stream-cursor-position stream)
-      (with-output-recording-options (stream :record t :draw nil)
-        (funcall continuation stream)
-        (force-output stream))
-      (setf (stream-cursor-position stream)
-            (values cursor-old-x cursor-old-y))
-      (with-output-recording-options (stream :record nil :draw nil)
-        (adjust-table-cells table stream)
-        (when multiple-columns (adjust-multiple-columns table stream))
-        (tree-recompute-extent table))
-      (replay table stream)
-      (if move-cursor
-          (setf (stream-cursor-position stream)
-                (values (bounding-rectangle-max-x table)
-                        (bounding-rectangle-max-y table)))
-          (setf (stream-cursor-position stream)
-                (values cursor-old-x cursor-old-y))))))
+  (apply #'invoke-with-new-output-record stream
+         (lambda (stream table)
+           (multiple-value-bind (cursor-old-x cursor-old-y)
+               (stream-cursor-position stream)
+             (with-output-recording-options (stream :record t :draw nil)
+               (funcall continuation stream)
+               (force-output stream))
+             (setf (stream-cursor-position stream)
+                   (values cursor-old-x cursor-old-y))
+             (with-output-recording-options (stream :record nil :draw nil)
+               (adjust-table-cells table stream)
+               (when multiple-columns (adjust-multiple-columns table stream))
+               (tree-recompute-extent table))
+             (replay table stream)
+             (setf (stream-cursor-position stream)
+                   (if move-cursor
+                       (values (bounding-rectangle-max-x table)
+                               (bounding-rectangle-max-y table))
+                       (values cursor-old-x cursor-old-y)))))
+         record-type
+         :x-spacing x-spacing
+         :y-spacing y-spacing
+         :multiple-columns multiple-columns
+         :multiple-columns-x-spacing multiple-columns-x-spacing
+         :equalize-column-widths equalize-column-widths
+         (alexandria:remove-from-plist
+          initargs :record-type :x-spacing :y-spacing :multiple-columns
+                   :multiple-columns-x-spacing :equalize-column-widths)))
```

Comment:
use the constant list of initargs (which are keywords).

*** 

Instead of removing intiargs, this should be

(with-keywords-removed (initargs (:record-type))
  (apply stream lambda record-type initargs))

because otherwise when some other method verifies, whether the argument was supplied (for whatever reason), it will always conclude that yes, it was.

Reply:
You mean after changing this to a generic function? This code would be the default method, right?

I assumed the argument processing was done to inject default values for unsupplied keywords which would work. But in the function lambda list, all keyword parameters default to `nil` if unsupplied, so I'm not sure what the intention was. Given that, I tend to agree that just leaving everything but `record-type` alone is probably best.

Reply:
when value needs processing, it needs to be passed explicitly - yes. for other cases, the default method should have its own defaults (and macro can have these defaults listed for a sake of editor hints when writing the macro).